### DISCUSSION TITLE
Transformation not working with logic written

### DISCUSSION THREAD

**Discussion by VivekSinghBi** [nextgenhealthcare/connect#5800] [Q&A] [answered]
Hi Team,

I have tried creating channel where input data is CSV and output would be progress note C-CDA.

I have been able to map all the demographics values, but I am getting error when processed on sample message.
--------------------------------------------------------------------------------------------------------------------------------------
Transformer error
ERROR MESSAGE: Error evaluating transformer
com.mirth.connect.server.MirthJavascriptTransformerException: 
CHANNEL:	CSV_C-CDAV3
CONNECTOR:	sourceConnector
SCRIPT SOURCE:	TRANSFORMER
SOURCE CODE:	
LINE NUMBER:	40984
DETAILS:	TypeError: Cannot read property "section" from undefined
	at 32bbe020-e5b8-4549-bd98-b79c33840a33_JavaScript_Filter_Transformer_0:40984 (doTransform)
	at 32bbe020-e5b8-4549-bd98-b79c33840a33_JavaScript_Filter_Transformer_0:41007 (doScript)
	at 32bbe020-e5b8-4549-bd98-b79c33840a33_JavaScript_Filter_Transformer_0:41009
	at com.mirth.connect.server.transformers.JavaScriptFilterTransformer$FilterTransformerTask.doCall(JavaScriptFilterTransformer.java:235)
	at com.mirth.connect.server.transformers.JavaScriptFilterTransformer$FilterTransformerTask.doCall(JavaScriptFilterTransformer.java:187)
	at com.mirth.connect.server.util.javascript.JavaScriptTask.call(JavaScriptTask.java:114)
	at java.base/java.util.concurrent.FutureTask.run(FutureTask.java:264)
	at java.base/java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1136)
	at java.base/java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:635)
	at java.base/java.lang.Thread.run(Thread.java:833)

**Comment by pacmano1** [comment]
If you don't post code, you rarely will get help here.

**Comment by jonbartels** [comment]
This should be moved to a Discussion.

Post the code or mappings that lead to this error please.

**Comment by VivekSinghBi** [comment]
Sorry, I am new. I used drag and drop functionality to map those values. I have shared below.
Any implementation guide would be helpful for the CCDA integration. 

/*GETTING PROPER MAPPING OF ALL THE DATA ELEMENTS INSIDE PROGRESS NOTES*/

tmp['recordTarget']['patientRole']['id'][0] = msg['row'][1]['patient_id'].toString();
tmp['recordTarget']['patientRole']['patient']['name']['given'][0] = msg['row'][1]['given_name'].toString();
tmp['recordTarget']['patientRole']['patient']['name']['family'] = msg['row'][1]['family_name'].toString();
tmp['recordTarget']['patientRole']['patient']['administrativeGenderCode']['@code'] = msg['row'][1]['sex_code'].toString();
tmp['recordTarget']['patientRole']['patient']['administrativeGenderCode']['@displayName'] = msg['row'][1]['gender'].toString();
tmp['recordTarget']['patientRole']['patient']['birthTime']['@value'] = msg['row'][1]['date_of_birth'].toString();
tmp['recordTarget']['patientRole']['addr']['streetAddressLine'] = msg['row'][1]['address_lines'].toString();
tmp['recordTarget']['patientRole']['addr']['city'] = msg['row'][1]['address_city'].toString();
tmp['recordTarget']['patientRole']['addr']['state'] = msg['row'][1]['address_state'].toString();
tmp['recordTarget']['patientRole']['addr']['postalCode'] = msg['row'][1]['postal_code'].toString();
tmp['informant'][0]['assignedEntity']['telecom'] = msg['row'][1]['telephone'].toString();
tmp['recordTarget']['patientRole']['providerOrganization']['id'] = msg['row'][1]['npi'].toString();
tmp['component']['structuredBody']['component'][0]['section']['code']['@code']=msg['row'][1]['LOINC_Code'].toString();
tmp['component']['structuredBody']['component'][0]['section']['title']=msg['row'][1]['Display_name'].toString();

**Reply by jonbartels** [reply, marked as answer]
The error is `DETAILS: TypeError: Cannot read property "section" from undefined`

Therefore the problem has to be on a line reading `section`:
```
tmp['component']['structuredBody']['component'][0]['section']['code']['https://github.com/code']=msg['row'][1]['LOINC_Code'].toString();
tmp['component']['structuredBody']['component'][0]['section']['title']=msg['row'][1]['Display_name'].toString();
```

You need to check if there are any components before trying to read the section.

**Reply by VivekSinghBi** [reply]
Thanks @jonbartels, could you help if we want to map those values with next sections of component.

**Comment by pacmano1** [comment]
@VivekSinghBi please do not post problems related to developing channels in Mirth to the "issues" section.   I have moved your question to where it should sit - in the Q and A section of discussions.

**Reply by VivekSinghBi** [reply]
Certainly, thank you @pacmano1
